packages/auth/src/accessToken/service.ts

```diff
@@ -72,16 +90,41 @@ async function introspect(
   }
 }
 
-async function revoke(
-  deps: ServiceDependencies,
-  id: string
-): Promise<Error | undefined> {
+async function revoke(deps: ServiceDependencies, id: string): Promise<void> {
   const token = await AccessToken.query(deps.knex).findById(id)
   if (token) {
-    if (!isTokenExpired(token)) {
-      await token.$query(deps.knex).patch({ expiresIn: 1 })
+    await token.$query(deps.knex).delete()
+  }
+}
+
+async function rotate(
+  deps: ServiceDependencies,
+  managementId: string
+): Promise<Rotation> {
+  let token = await AccessToken.query(deps.knex).findOne({ managementId })
+  if (token) {
+    await token.$query(deps.knex).delete()
+    const newManagementId = uuid()
+    token = await AccessToken.query(deps.knex).insertAndFetch({
+      value: crypto.randomBytes(8).toString('hex').toUpperCase(),
+      grantId: token.grantId,
+      expiresIn: token.expiresIn,
+      managementId: newManagementId
+    })
+    const access = await Access.query(deps.knex).findOne({
+      grantId: token.grantId
+    })
```

Comment:
There could be multiple accesses attached to that grant / token. So please use `where`.